cs_volume.py

```diff
@@ -0,0 +1,749 @@
+                if 'errortext' in volume:
+                    self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+                poll_async = self.module.params.get('poll_async')
+                if poll_async:
+                    volume = self.poll_job(res, 'volume')
+        return volume
+
+    def absent_volume(self):
+        volume = self.get_volume()
+
+        if volume:
+            self.result['changed'] = True
+            if not self.module.check_mode:
+                volume = self.detached_volume()
+
+                res = self.cs.deleteVolume(id=volume['id'])
+                if 'errortext' in volume:
+                    self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+                poll_async = self.module.params.get('poll_async')
+                if poll_async:
+                    res = self.poll_job(res, 'volume')
+
+        return volume
+
+    def get_result(self, volume):
+        if volume:
+            if 'id' in volume:
+                self.result['id'] = volume['id']
+            if 'name' in volume:
+                self.result['name'] = volume['name']
+            if 'displayname' in volume:
+                self.result['display_name'] = volume['displayname']
+            if 'group' in volume:
+                self.result['group'] = volume['group']
+            if 'domain' in volume:
+                self.result['domain'] = volume['domain']
+            if 'account' in volume:
+                self.result['account'] = volume['account']
+            if 'project' in volume:
+                self.result['project'] = volume['project']
+            if 'zonename' in volume:
+                self.result['zone'] = volume['zonename']
+            if 'created' in volume:
+                self.result['created'] = volume['created']
+            if 'attached' in volume:
+                self.result['attached'] = volume['attached']
+            if 'virtualmachineid' in volume:
+                self.result['vm_id'] = volume['virtualmachineid']
+            if 'vmdisplayname' in volume:
+                self.result['vm_display_name'] = volume['vmdisplayname']
+            if 'state' in volume:
+                self.result['state'] = volume['state']
+            if 'deviceid' in volume:
+                self.result['device_id'] = volume['deviceid']
+        return self.result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(required=True),
+            account = dict(default=None),
+            custom_id = dict(default=None),
+            disk_offering = dict(default=None),
+            display_volume = dict(choices=BOOLEANS, default=False),
+            domain = dict(default=None),
+            max_iops = dict(type='int', default=None),
+            min_iops = dict(type='int', default=None),
+            project = dict(default=None),
+            size = dict(type='int', default=None),
+            snapshot_id = dict(default=None),
```

Comment:
didn't find anything about snapshot_id in the code. seems to be unused. I am fine to add this functionality later if you like